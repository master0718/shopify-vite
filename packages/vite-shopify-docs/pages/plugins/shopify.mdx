[View this package on NPM.](https://www.npmjs.com/package/vite-plugin-shopify)

The core of this project is `vite-plugin-shopify`, which allows Shopify theme developers to use Vite as their front-end build tool. This plugin has everything you need to take advantage of all the great features of Vite when developing Shopify themes. More details can be found in the plugin README [here](https://github.com/barrel/barrel-shopify/blob/main/packages/vite-plugin-shopify/README.md).
