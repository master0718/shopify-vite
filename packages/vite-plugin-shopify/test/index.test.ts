import { describe, expect, it } from 'vitest'
import path from 'path'
import { build } from 'vite'
import shopify from '../src'
import fs from 'fs/promises'

describe('vite-plugin-shopify', () => {
  it('runs as expected', async () => {
    await build({
      logLevel: 'silent',
      plugins: [
        shopify({
          themeRoot: path.join(__dirname, '__fixtures__'),
          sourceCodeDir: path.join(__dirname, '__fixtures__', 'frontend')
        })
      ]
    })

    const tagsHtml = await fs.readFile(path.join(__dirname, '__fixtures__', 'snippets', 'vite-tag.liquid'), { encoding: 'utf8' })
    const clientHtml = await fs.readFile(path.join(__dirname, '__fixtures__', 'snippets', 'vite-client.liquid'), { encoding: 'utf8' })

    expect(tagsHtml).toMatchSnapshot()
    expect(clientHtml).toMatchSnapshot()
  })
})
