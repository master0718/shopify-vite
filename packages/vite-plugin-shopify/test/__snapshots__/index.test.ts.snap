// Vitest Snapshot v1

exports[`vite-plugin-shopify > builds out .liquid files for production 1`] = `
"{% comment %}
  IMPORTANT: This snippet is automatically generated by vite-plugin-shopify.
  Do not attempt to modify this file directly, as any changes will be overwritten by the next build.
{% endcomment %}
{% assign path = vite-tag | replace: '@@/', '../../resources/js/' | replace: '~/', '../' | replace: '@/', '../' %}
{% if path == \\"/test/__fixtures__/frontend/entrypoints/theme.css\\" or path == \\"theme.css\\" %}
  {{ 'theme-1a84b608.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload_stylesheet }}
{% elsif path == \\"/test/__fixtures__/frontend/entrypoints/theme.ts\\" or path == \\"theme.ts\\" %}
  <script src=\\"{{ 'theme-ec808ebb.js' | asset_url | split: '?' | first }}\\" type=\\"module\\" crossorigin=\\"anonymous\\"></script>
{% endif %}
"
`;
